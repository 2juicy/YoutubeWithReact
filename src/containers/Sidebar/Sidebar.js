import React from "react";
import { Menu, Divider } from "semantic-ui-react";
import "./Sidebar.scss";
import SidebarItem from "./SidebarItem/SidebarItem";
import SidebarHeader from "./SidebarHeader/SidebarHeader";
import Subscriptions from "./Subscriptions/Subscriptions";
import SidebarFooter from "./SidebarFooter/SidebarFooter";

export default function Sidebar() {
  return (
    <Menu borderless vertical stackable fixed="left" className="side-nav">
      <SidebarItem highlight label="Home" icon="home" />
      <SidebarItem label="Trending" icon="fire" />
      <SidebarItem label="Followers" icon="spy" />
      <Divider />
      <SidebarHeader title="Library" />
      <SidebarItem label="History" icon="history" />
      <SidebarItem label="Watch later" icon="clock" />
      <SidebarItem label="Liked videos" icon="thumbs up" />
      <Divider />
      <Subscriptions />
      <SidebarHeader title="More from YourTube" />
      <SidebarItem label="Movies and Shows" icon="film" />
      <Divider />
      <SidebarItem label="Report History" icon="flag" />
      <SidebarItem label="Help" icon="help circle" />
      <SidebarItem label="Send feedback" icon="comment" />
      <Divider />
      <SidebarFooter />
    </Menu>
  );
}
