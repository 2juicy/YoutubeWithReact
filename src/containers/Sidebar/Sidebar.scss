.ui.menu.fixed.side-nav {
  background-color: #f5f5f5;
  margin-top: 62px;
  padding-bottom: 62px;
  overflow-y: auto;
}

.side-nav.ui.vertical.menu {
  width: 17rem;
}
