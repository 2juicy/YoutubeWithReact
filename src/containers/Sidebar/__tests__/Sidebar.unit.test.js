import React from "react";
import { shallow } from "enzyme";
import Sidebar from "../Sidebar";

describe("Sidebar", () => {
  test("renders", () => {
    const wrapper = shallow(<Sidebar />);
    expect(wrapper).toMatchSnapshot();
  });
});
