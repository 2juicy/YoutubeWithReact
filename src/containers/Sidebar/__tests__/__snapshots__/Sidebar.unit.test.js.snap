// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders 1`] = `
<Menu
  borderless={true}
  className="side-nav"
  fixed="left"
  stackable={true}
  vertical={true}
>
  <SidebarItem
    highlight={true}
    icon="home"
    label="Home"
  />
  <SidebarItem
    icon="fire"
    label="Trending"
  />
  <SidebarItem
    icon="spy"
    label="Followers"
  />
  <Divider />
  <SidebarHeader
    title="Library"
  />
  <SidebarItem
    icon="history"
    label="History"
  />
  <SidebarItem
    icon="clock"
    label="Watch later"
  />
  <SidebarItem
    icon="thumbs up"
    label="Liked videos"
  />
  <Divider />
  <Subscriptions />
  <SidebarHeader
    title="More from YourTube"
  />
  <SidebarItem
    icon="film"
    label="Movies and Shows"
  />
  <Divider />
  <SidebarItem
    icon="flag"
    label="Report History"
  />
  <SidebarItem
    icon="help circle"
    label="Help"
  />
  <SidebarItem
    icon="comment"
    label="Send feedback"
  />
  <Divider />
  <SidebarFooter />
</Menu>
`;
